workflows/prognostic_c48_run/runtime/steppers/radiation.py

```diff
@@ -0,0 +1,221 @@
+        TODO: make scheme able to read existing forcing; make lookup data part of
+        writing a run directory?
+        """
+        if self._comm.rank == 0:
+            for target, local in zip(
+                (lookup_data_path, forcing_data_path),
+                (lookup_local_dir, forcing_local_dir),
+            ):
+                io.get_remote_tar_data(target, local)
+        self._comm.barrier()
+        self._lookup_local_dir = lookup_local_dir
+        self._forcing_local_dir = forcing_local_dir
+
+    def _init_driver(self, fv_core_dir: str = "./INPUT/"):
+        """Initialize the radiation driver"""
+        sigma = io.load_sigma(fv_core_dir)
+        nlay = len(sigma) - 1
+        aerosol_data = io.load_aerosol(self._forcing_local_dir)
+        sfc_filename, sfc_data = io.load_sfc(self._forcing_local_dir)
+        solar_filename, _ = io.load_astronomy(
+            self._forcing_local_dir, self._rad_config["isolar"]
+        )
+        self._driver.radinit(
+            sigma,
+            nlay,
+            self._rad_config["imp_physics"],
+            self._comm.rank,
+            self._rad_config["iemsflg"],
+            self._rad_config["ioznflg"],
+            self._rad_config["ictmflg"],
+            self._rad_config["isolar"],
+            self._rad_config["ico2flg"],
+            self._rad_config["iaerflg"],
+            self._rad_config["ialbflg"],
+            self._rad_config["icldflg"],
+            self._rad_config["ivflip"],
+            self._rad_config["iovrsw"],
+            self._rad_config["iovrlw"],
+            self._rad_config["isubcsw"],
+            self._rad_config["isubclw"],
+            self._rad_config["lcrick"],
+            self._rad_config["lcnorm"],
+            self._rad_config["lnoprec"],
+            self._rad_config["iswcliq"],
+            aerosol_data,
+            solar_filename,
+            sfc_filename,
+            sfc_data,
+        )
+
+    def __call__(
+        self, time: cftime.DatetimeJulian, state: State,
+    ):
+        self._rad_update(time, self._timestep)
+        if self._input_generator is not None:
+            state = self._generate_inputs(state, time)
+        diagnostics = self._rad_compute(state, time)
+        return {}, diagnostics, {}
+
+    def _rad_update(self, time: cftime.DatetimeJulian, dt_atmos: float) -> None:
+        """Update the radiation driver's time-varying parameters"""
+        # idat is supposed to be model initalization time but is unused w/ current flags
+        idat = np.array(
+            [time.year, time.month, time.day, 0, time.hour, time.minute, time.second, 0]
+        )
+        jdat = np.array(
+            [time.year, time.month, time.day, 0, time.hour, time.minute, time.second, 0]
+        )
+        fhswr = np.array(float(self._rad_config["fhswr"]))
+        dt_atmos = np.array(float(dt_atmos))
+        aerosol_data = io.load_aerosol(self._forcing_local_dir)
```

Comment:
Looks like this I/O happens every time step? Should you move this to the initialization?

Reply:
As discussed will move this

Reply:
Done